design-documents/asynchronous-operations/poison-pill.md

```diff
@@ -0,0 +1,20 @@
+## Glossary
+
+poison pill
+: message in Data Storage that signals that consumer should die
+
+### Overview
+
+Right now our consumer depends on config/objects in-memory state. After application was bootstrapped it initialize a lot of object that are always kept in memory after first load - config, websites, etc.., since our consumers are long-living and their live time limited only by message count it leads to outdated im-memory config/objects states and we need to introduce the way to reinit them.
+
+### Design
+
+Before process new message consumer should ask Data Storage if there are poison pill for him. If yes, consumer dies and next cron:run will start new instance of consumer.
```

Comment:
If there is a possibility that several consumers would be started simultaneously, how are you going to deliver your poison pill for each of them?

Reply:
Each of them will compare version of poison pill in his memory vs latest in DB